core/Rename.cpp

```diff
@@ -122,20 +134,27 @@ namespace olympia
         if(dests.size() > 0)
         {
             sparta_assert(dests.size() == 1); // we should only have one destination
-            auto const & original_dest = inst_ptr->getRenameData().getOriginalDestination();
-            --reference_counter_[original_dest.rf][original_dest.val];
-            // free previous PRF mapping if no references from srcs, there should be a new dest mapping for the ARF -> PRF
-            // so we know it's free to be pushed to freelist if it has no other src references
-            if(reference_counter_[original_dest.rf][original_dest.val] <= 0){
-                freelist_[original_dest.rf].push(original_dest.val);
+            const auto dest = dests[0];
+            const auto rf  = olympia::coreutils::determineRegisterFile(dest);
+            const auto num = dest.field_value;
+            const bool is_x0 = num != 0 || rf != core_types::RF_INTEGER;
```

Comment:
My brain hurts reading this...  do you mean `is_not_x0`?

Reversed logic:
```suggestion
            const bool is_x0 = num == 0 && rf != core_types::RF_INTEGER;
            if (false == is_x0)
```